waku/v2/protocol/waku_filter/waku_filter.nim

```diff
@@ -208,26 +208,47 @@ proc subscription*(proto: WakuFilter): MessageNotificationSubscription =
       for filter in subscriber.filter.contentFilters:
         if msg.contentTopic in filter.topics:
           let push = FilterRPC(requestId: subscriber.requestId, push: MessagePush(messages: @[msg]))
-          let conn = await proto.switch.dial(subscriber.peer.peerId, subscriber.peer.addrs, WakuFilterCodec)
-          await conn.writeLP(push.encode().buffer)
+          
+          let connOpt = await proto.peerManager.dialPeer(subscriber.peer, WakuFilterCodec)
+
+          if connOpt.isSome:
+            await connOpt.get().writeLP(push.encode().buffer)
+          else:
+            # @TODO more sophisticated error handling here
+            error "failed to push messages to remote peer"
+            waku_filter_errors.inc(labelValues = ["dial_failure"])
```

Comment:
A suggestion, you may define failure messages like "dial_failure" as constant variables  and then keep using those variables in the code.

Reply:
Thanks, good idea. I generally hate magic variables and this way it's easier to keep track of them in one place and avoid errors. Done in 4e7f068